### ISSUE TITLE
Loading invalid post-data of post_id==0 on WP-Admin Login

### ISSUE THREAD

**Issue by bgoeschi** [sybrew/the-seo-framework#482]
On the wp-admin login I get the following warnings:
`Warning: array_intersect_key(): Argument #1 is not an array in /home/.sites/265/site6623448/web/wp-content/plugins/autodescription/inc/classes/post-data.class.php on line 116`
`Warning: Invalid argument supplied for foreach() in /home/.sites/265/site6623448/web/wp-content/plugins/autodescription/inc/classes/post-data.class.php on line 120`
and
`Warning: array_merge(): Argument #2 is not an array in /home/.sites/265/site6623448/web/wp-content/plugins/autodescription/inc/classes/post-data.class.php on line 124`

I inserted a few lines of debug code and found that it happens whenever `$post_id==0` and that it somehow loads a post object that is filled with a lot of garbage:
`WP_Post Object ([ID] => 0 [post_author] => 0 [post_date] => 0000-00-00 00:00:00 [post_date_gmt] => 0000-00-00 00:00:00 [post_content] => [theme-my-login action="login"] [post_title] => Anmelden [post_excerpt] => [post_status] => publish [comment_status] => closed [ping_status] => closed [post_password] => [post_name] => login [to_ping] => [pinged] => [post_modified] => 0000-00-00 00:00:00 [post_modified_gmt] => 0000-00-00 00:00:00 [post_content_filtered] => [post_parent] => 0 [guid] => [menu_order] => 0 [post_type] => page [post_mime_type] => [comment_count] => 0 [filter] => raw )`

For me the quick-and-dirty-fix is to check for the id around line 100 in inc/classes/post-data.class.php:
```		// get_post_meta() requires a valid post ID. Make sure that post exists.
		$post = \get_post( $post_id );

		if ( ! $post or $post_id == 0)
			return $cache[ $post_id ] = [];
```
After adding `or $post_id == 0`, the warnings disappeared.
Some better integrity checks on the post object would probably be a better solution though.

**Comment by bgoeschi** [comment]
It looks like the post object is coming from the [Theme My Login](https://github.com/theme-my-login/theme-my-login) plugin.

**Comment by sybrew** [comment]
Post ID 0 can't exist by default, and shouldn't exist by non-default. Likely this is because of something building a CMS on top of a CMS... 🤕 

Related: #479.
Addressed in: 189bcc7eaa38b1ab26cec7163a714a965f96bd62.
